blockchain/message_pool/src/msgpool.rs

```diff
@@ -328,24 +325,32 @@ where
 
     /// Push a signed message to the MessagePool
     pub async fn push(&self, msg: SignedMessage) -> Result<Cid, Error> {
+        self.check_message(&msg).await?;
         let cid = msg.cid().map_err(|err| Error::Other(err.to_string()))?;
-        self.add(&msg).await?;
+        self.add_tipset(msg.clone(), &self.cur_tipset.read().await.clone())
+            .await?;
         self.add_local(msg).await?;
+        // TODO: Publish over Gossip
         Ok(cid)
     }
 
-    /// This is a helper to push that will help to make sure that the message fits the parameters
-    /// to be pushed to the MessagePool
-    pub async fn add(&self, msg: &SignedMessage) -> Result<(), Error> {
-        let size = msg.marshal_cbor()?.len();
-        if size > 32 * 1024 {
+    /// Basic checks on the validity of a message
+    async fn check_message(&self, msg: &SignedMessage) -> Result<(), Error> {
+        if msg.marshal_cbor()?.len() > 32 * 1024 {
             return Err(Error::MessageTooBig);
         }
-        if msg.value() > &BigInt::from(2_000_000_000u64) {
+        msg.valid_for_block_inclusion(0).map_err(Error::Other)?;
+        if msg.value() > &BigInt::from(types::TOTAL_FILECOIN) {
             return Err(Error::MessageValueTooHigh);
         }
```

Comment:
Looks like this check for the gas free cap is missing. In Lotus they have : 

	if m.Message.GasFeeCap.LessThan(minimumBaseFee) {
		return ErrGasFeeCapTooLow
	}

Reply:
Just noticed your comment below. Dont see why this couldn't be done now tho